lib/middleware/utils.ts

```diff
@@ -0,0 +1,28 @@
+type TargetCommandMap = {
+  [key: string]: {
+    target?: boolean;
+    subCommands?: TargetCommandMap;
+  };
+};
+
+export function isTargetedCommand(
```

Comment:
I made a few small changes to this. I removed the nested func b/c it wasn't necessary for the recursion, and I also changed the input to an array.

This will make it easier for me to use in other places (I'll need it for the Fire Alarm work)